src/Zicht/Bundle/UrlBundle/Url/DelegatingProvider.php

```diff
@@ -87,16 +90,24 @@ public function suggest($pattern)
     }
 
     /**
-     * @{inheritDoc}
+     * {@inheritDoc}
      */
     public function all(AuthorizationCheckerInterface $securityContext)
     {
         $ret = array();
-        foreach ($this->providers as $provider) {
+        foreach ($this->getProviders() as $provider) {
             if ($provider instanceof ListableProvider) {
                 $ret = array_merge($ret, $provider->all($securityContext));
             }
         }
         return $ret;
     }
+
+    /**
+     * @return Provider[]|array
+     */
+    private function getProviders()
+    {
+        return iterable($this->providers)->sorted('priority')->map('provider')->values();
```

Comment:
this dependency is removed, prefereer to do this without iterable and revert back to old code and keep depenedncies as little a possible. This also can lead to performance issues because eveytime one of the public function is called it has to do the whole sorting, mapping and extracting again.